composition-js/src/merging/merge.ts

```diff
@@ -1810,85 +1863,86 @@ class Merger {
     //   definition is the intersection of all definitions (meaning that if there divergence in
     //   locations, we only expose locations that are common everywhere).
     this.mergeDescription(sources, dest);
-    if (sources.some((s) => s && this.isMergedDirective(s))) {
+
+    const definedSources = sources.filter((src): src is DirectiveDefinition => src !== undefined);
+    if (definedSources.some(s => s.locations.some(loc => executableDirectiveLocations.includes(loc)))) {
       this.mergeExecutableDirectiveDefinition(sources, dest);
+    } else {
+      if (this.isMergedDirective(dest)) {
```

Comment:
**question** Is there a reason to keep this as an `if` nested under `else` or could this be an `else if`?